tests/phpunit/tests/webfonts-api/wpWebfontsRegistry.php

```diff
@@ -0,0 +1,337 @@
+		$this->validator_mock
+			->expects( $this->once() )
+			->method( 'is_valid_schema' )
+			->willReturn( true );
+		$this->validator_mock
+			->expects( $this->once() )
+			->method( 'set_valid_properties' )
+			->willReturn( $validated_webfont );
+
+		$this->assertSame( $expected, $this->registry->register( $webfont ) );
+	}
+
+	/**
+	 * Data Provider.
+	 *
+	 * return @array
+	 */
+	public function data_register_with_valid_schema() {
+		return array(
+			'valid schema without font-display' => array(
+				'webfont'           => array(
+					'provider'    => 'google',
+					'font_family' => 'Roboto',
+					'font_style'  => 'normal',
+					'font_weight' => 'normal',
+				),
+				'validated_webfont' => array(
+					'provider'     => 'google',
+					'font-family'  => 'Roboto',
+					'font-style'   => 'normal',
+					'font-weight'  => 'normal',
+					'font-display' => 'fallback',
+				),
+				'expected'          => 'roboto.normal.normal',
+			),
+			'valid schema with src'             => array(
+				'webfont'           => array(
+					'provider'     => 'local',
+					'font_family'  => 'Source Serif Pro',
+					'font_style'   => 'normal',
+					'font_weight'  => '200 900',
+					'font_stretch' => 'normal',
+					'src'          => 'https://example.com/assets/fonts/source-serif-pro/SourceSerif4Variable-Roman.ttf.woff2',
+				),
+				'validated_webfont' => array(
+					'provider'     => 'local',
+					'font-family'  => 'Source Serif Pro',
+					'font-style'   => 'normal',
+					'font-weight'  => '200 900',
+					'font-display' => 'fallback',
+					'font-stretch' => 'normal',
+					'src'          => 'https://example.com/assets/fonts/source-serif-pro/SourceSerif4Variable-Roman.ttf.woff2',
+				),
+				'expected'          => 'source-serif-pro.normal.200 900',
+			),
+		);
+	}
+
+	/**
+	 * @covers WP_Webfonts_Registry::get_by_provider
+	 */
+	public function test_get_by_provider_when_does_not_exist() {
+		/*
+		 * Set the `registry_by_provider` property.
+		 * This is set in WP_Webfonts_Registry::register(), which not part of this test.
+		 */
+		$property = new ReflectionProperty( $this->registry, 'registry_by_provider' );
+		$property->setAccessible( true );
+		$property->setValue( $this->registry, array( 'google', 'local' ) );
+
+		$this->assertSame( array(), $this->registry->get_by_provider( 'my-custom-provider' ) );
+	}
+
+	/**
+	 * As there are many moving parts to getting by provider, this test is an integration
+	 * test that does not mock.
+	 *
+	 * @covers WP_Webfonts_Registry::get_by_provider
+	 * @covers WP_Webfonts_Registry::register
+	 *
+	 * @dataProvider data_get_by_provider_integrated
+	 *
+	 * @param array  $webfonts    Given webfont to register.
+	 * @param string $provider_id Provider ID to query.
+	 * @param array  $expected    Expected return value.
+	 */
+	public function test_get_by_provider_integrated( array $webfonts, $provider_id, $expected ) {
```

Comment:
```suggestion
	public function test_get_by_provider_integrated( array $webfonts, $provider_id, array $expected ) {
```
`$expected` is documented as an array.

Reply:
Thank you @costdev - fixed in https://github.com/WordPress/wordpress-develop/pull/1736/commits/92bd91195f7896404e666f8a091b7224903157e9.